# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/generic/nsBlockFrame.h
# Commit: b2727a598452
# Full Hash: b2727a598452fb942e0611df5f931d4df0efaceb
# Author: Frédéric Wang <fwang@igalia.com>
# Date: 2024-01-18 04:47:51
# Regressor Bug: 1765615
# File Overlap Count: 1
# Description:
#   Bug 1765615 - Handle most changes to CSS `contain` and `content-visibility` without needing to reconstruct frames. r=layout-reviewers,emilio
#   
#   Right now, we reconstruct frames in response to a change in the CSS
#   `contain` property or `content-visibility`. This patch tries to optimize
#   this a bit:
# ==============================================================================

diff -r 5608664350d8 -r b2727a598452 layout/generic/nsBlockFrame.h
--- a/layout/generic/nsBlockFrame.h	Wed Jan 17 07:22:29 2024 +0000
+++ b/layout/generic/nsBlockFrame.h	Wed Jan 17 08:22:48 2024 +0000
@@ -654,13 +654,6 @@
    */
   bool IsInLineClampContext() const;
 
- protected:
-  /** grab overflow lines from this block's prevInFlow, and make them
-   * part of this block's mLines list.
-   * @return true if any lines were drained.
-   */
-  bool DrainOverflowLines();
-
   /**
    * @return false iff this block does not have a float on any child list.
    * This function is O(1).
@@ -681,6 +674,21 @@
   }
 
   /**
+   * @return true if NS_BLOCK_DYNAMIC_BFC should be set on this frame.
+   */
+  bool IsDynamicBFC() const {
+    return StyleDisplay()->IsContainPaint() ||
+           StyleDisplay()->IsContainLayout();
+  }
+
+ protected:
+  /** grab overflow lines from this block's prevInFlow, and make them
+   * part of this block's mLines list.
+   * @return true if any lines were drained.
+   */
+  bool DrainOverflowLines();
+
+  /**
    * Moves frames from our PushedFloats list back into our mFloats list.
    */
   void DrainSelfPushedFloats();